### ISSUE TITLE
There is no fine granularity for Postman variables, missing request scope and scenarios

### ISSUE THREAD

**Issue by antonalechnovic** [a85/POSTMan-Chrome-Extension#5357]
**Is your feature request related to a problem? Please describe.**
Scenario: public API with single entry point. Complex POST request to send JSON. Requests to the same path can be different based on "type" JSON field. Different authentication details with different payload and different type of secret (so need to test across users as well).  Parameterization requirement is across several dimensions: 
1. ability to switch user (e.g. JWT) , 
2. ability to switch environment (e.g. URL)
3. ability to switch JSON body payload fields

Ideally I would like to have flexibility to switch all 3 without creating duplicate requests and without abusing/polluting global and environmental variables. 

With "Postman environments and environmental variables" I can solve problem 2. 
By creating two collections with almost identical request I can solve problem 3. (even without using variables, just hardcoding JSON payload, but that sucks because now I have duplicate request, so maintaining the changes to API becomes a hassle.

How to tackle user switch(problem 1) no clue, yes I could create two envs PROD_BOB; PROD_ALICE to flatten the structure and add both user details and prod enviornment url details to single environment but that will eventually become unmanageable.. 

**Describe the solution you'd like**
Part 1: First of all it would be nice to have "Request Scope" (configurable at the request level, ATM you don't provide that for "raw" JSON payload) with UI tab with visible parameters which are only applied at request level. This is useful when sharing request with clients as:

A. variables in request scope are self contained in the request so don't pollute env and global scopes.
B. it is immediately visible what they have to change, without dummy data and without ugly placeholders like `<enter your stuff here>`.

 The only way to do something similar right now is to parameterize request with placeholders and then use pre-request script section to set pm variables. But that is bad as now you glue based on field names so maintenance becomes hassle again, and since you **forgot?!** to create `pm.request.set ` you still pollute environment and global scopes with `pm.environment.set and pm.globals.set`

Part 2:
Baby-sitting me with scopes is all nice. But I want more :-) I want a concept of scenarios, where I could say:
1. My system environments for selection are: "prod", "stage", "dev"
2. My auth user details for selection are : "bob" , "alice", "tracey"
3. My x  details for selection are : "a", "b" , "c"

My scenario is "Majestic" (1. "stage", 2. "bob", 3. "a")
My other scenario is "Brutal" (1. "prod", 2. "tracey", 3. "c")

If you take these two scenarios the requests are actually identical, but currently you cannot achieve anything like that without making your environments look like pile of garbage/denormalized db.

Basically I want an hierarchy of environments(e.g. env1 contains env2 and env3) with a rule that env2 cannot contain same variables as env3, as overriding can happen only from parent to child.

**Comment by gekonwi** [comment]
The idea of "Scenarios" is genius!! Just like a row in a CSV file, but named and available as a drop-down in the app (e.g. right next to the Environments drop-down).

We also often have the need to establish relationships between variables. E.g. our API supports different types of payload. It is the same XML request body, but the content of the <payload> tag can be plain text, like:
```
<payload>{{message}}<payload>
```  
Or pipe delimited, like:
```
<payload>{{message}}|{{priority}}|{{callBackNumber}}<payload>
```

The payload type is defined separately, as a URL parameter, like:
```
{{url}}/wctp/{{payloadType}}
```

We also support 4 different recipient types. 3 payload formats and 4 recipient types makes 12 combinations. Expressing each as an environment is not an option either - we already have 10 environments for the 10 different release versions we support (we work in the medical field, hospitals don't update as fast as we'd like). If we were to flatten the three dimensions (release, recipient type, payload type) into environments, we would end up with 120 environments, which is unmanageable. 

Thus, we end up:
1. duplicating each request 12 times
2. polluting global variables (making it impossible to share collections across workspaces because the other workspaces may not have the right global variables)

If we had the concept of "Scenarios", we could use one request for each request type, our 10 environments, and 12 Scenarios (one for each combination of the 4 recipient types with the 3 payload types).

**Comment by a85** [comment]
Postman supports collection variables now that should help with these. Let me know if this doesn't work and I'll re-open to investigate more.

**Comment by airtonix** [comment]
to be honest, if the postman team just wrapped newman around jest/tape/mocha  then all these problems would be a non issue.

i'd be able to do:

```

beforeEach(() => {
  //set for this endpoint
})

afterEach(() => {
  //teardown for this endpoint
})

it('requests with default body', () => {
  postman.request() // implicitly calls this endpoint
})

it('requests with something else', async () => {
  const response = await postman.request({ 
    body: postman.getVariable('ExampleOfBrokenPayload')
  })
  expect(response.status).to.be(400)
})


it('requests with something different again', async () => {
  const response = await postman.request({
    body: postman.getVariable('ExampleOfYetAnotherBrokenPayload')
  })
  expect(response.status).to.be(400)
})
```

Instead, we have this weird myopic testing approach.
